postem/postem-app/src/main/java/org/sakaiproject/postem/controller/MainController.java

```diff
@@ -0,0 +1,114 @@
+/****************************************************************************** 
+* Copyright (c) 2020 Apereo Foundation
+
+* Licensed under the Educational Community License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+
+*          http://opensource.org/licenses/ecl2
+
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+ ******************************************************************************/
+package org.sakaiproject.postem.controller;
+
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.sakaiproject.api.app.postem.data.Gradebook;
+import org.sakaiproject.postem.constants.PostemToolConstants;
+import org.sakaiproject.postem.form.GradebookForm;
+import org.sakaiproject.postem.service.PostemSakaiService;
+import org.sakaiproject.tool.api.SessionManager;
+import org.sakaiproject.tool.api.ToolSession;
+import org.sakaiproject.tool.cover.ToolManager;
+import org.sakaiproject.user.api.PreferencesService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.servlet.LocaleResolver;
+import org.springframework.web.servlet.support.RequestContextUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.sakaiproject.component.cover.ServerConfigurationService;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Controller
+public class MainController {
+
+    @Autowired
+    private PostemSakaiService postemSakaiService;
+	
+    @Autowired
+    private SessionManager sessionManager;
+    
+    @Autowired
+    private PreferencesService preferencesService;
+
+    @GetMapping(value = {"/", "/index"})
+    public String showIndex(Model model, HttpServletRequest request, HttpServletResponse response) {
+        log.debug("showIndex");
+        
+        String userId = sessionManager.getCurrentSessionUserId();
+        final Locale locale = StringUtils.isNotBlank(userId) ? preferencesService.getLocale(userId) : Locale.getDefault();
```

Comment:
I would literally just add this logic to BasePreferencesService. If you pass a blank userId to the getLocale call, just return Locale.getDefault from the service. I think there's a lot of the pattern you're using here across Sakai, so you'd be doing people a favour just fixing preference service. Make sure to update the javadoc for PreferencesService.getLocale to reflect your change.

Reply:
Hi, Adrian;
I have made the change and updating the javadoc.
Thank you.